### PR TITLE
Move all healthcheck-related code to pkg/healthcheck

### PR DESCRIPTION
**Pull request by klingerf**
This branch is a substantial refactor to the way our health check code was setup:

* All checks are now defined in `pkg/healthcheck/healthcheck.go`
* Check definitions use the `healthcheck.checkers` struct instead of protobuf
* Checks can now be marked as fatal, which will skip all remaining checks on failure
* Checks are now binary (passing/failing); we no longer distinguish between failing and errored
* The `MockKubeApi` struct has been removed, since it was no longer used
* The `KubernetesApi` interface has been removed, since it only had one implementation after removing the mock

This refactor is in support of #1470, #1471, #1473, #1477, and #1478.

### CODE REVIEW

File: cli/cmd/check.go [hunk 1]
```diff
@@ -44,33 +36,7 @@ local system, the Linkerd control plane, and connectivity between those. The pro
...
-			}
-
-			grpcStatusChecker := healthcheck.NewGrpcStatusChecker(apiClient)
-			versionStatusChecker := version.NewVersionStatusChecker(versionCheckURL, options.versionOverride, apiClient)
-
-			err = checkStatus(os.Stdout, kubeApi, grpcStatusChecker, versionStatusChecker)
-			if err != nil {
-				os.Exit(2)
-			}
+			configureAndRunChecks(options)
```
Comment (dadjeibaah) [inline cli/cmd/check.go]: I know this isn't part of the change, but the use of the possessive pronoun "your" is a little jarring especially when compared to other command descriptions. We may want to revisit this in a new PR.

File: test/testdata/check.golden [hunk 2]
```diff
@@ -1,9 +1,12 @@
 kubernetes-api: can initialize the client..................................[ok]
 kubernetes-api: can query the Kubernetes API...............................[ok]
 kubernetes-api: is running the minimum Kubernetes API version..............[ok]
-linkerd-api[namespace]: control plane namespace exists.....................[ok]
+linkerd-api: control plane namespace exists................................[ok]
+linkerd-api: can initialize the client.....................................[ok]
+linkerd-api: can query the control plane API...............................[ok]
 linkerd-api[kubernetes]: control plane can talk to Kubernetes..............[ok]
 linkerd-api[prometheus]: control plane can talk to Prometheus..............[ok]
+linkerd-version: can get the latest version................................[ok]
```
Comment (rmars) [inline test/testdata/check.golden]: slight naming nit: can we use "can query for the latest version" or "can determine the latest version" instead? "can get" to me implies some downloading (I wasn't sure what it meant at first)

File: cli/cmd/check.go (hunk 1 above)
Comment (klingerf) [reply cli/cmd/check.go]: Ahhh, yeah, this is great feedback. I partially rewrote the description for this command in #1495, and I'll address this suggestion there.

As far as printing the skipped check goes, I'm a bit on the fence. If the user fails a fatal check, then they must fix it before being able to successfully use the CLI. In that case, printing the checks that were skipped might distract from the failure that requires fixing.

File: test/testdata/check.golden (hunk 2 above)
Comment (klingerf) [reply test/testdata/check.golden]: Ah, yep, great call! I like "can determine the latest version" a lot -- will update.

**Pull request merged**